llvm/include/llvm/Target/TargetSelectionDAG.td

```diff
@@ -1565,6 +1573,21 @@ def atomic_load_32 :
   let IsAtomic = 1;
   let MemoryVT = i32;
 }
+
+def atomic_load_iptr :
+  PatFrag<(ops node:$ptr),
+          (new_atomic_load node:$ptr)> {
+  let IsAtomic = 1;
+  let MemoryVT = i32;
+}
+
+def atomic_load_vtany32 :
```

Comment:
Should not need a separate set of atomic nodes. The memory type is really a lie, anything relying on this now should probably already be checking the size